src/vipersci/carto/heatmap.py

```diff
@@ -214,12 +192,11 @@ def as_ndarray(input: Sequence) -> NDArray:
 
     if transform is None:
         if sample_bounds is None:
-            minx = np.amin(x_coords_np)
-            maxy = np.amax(y_coords_np)
+            sample_bounds = compute_bounds(x_coords_np, y_coords_np)
         else:
-            minx = sample_bounds.bounds[0]
-            maxy = sample_bounds.bounds[3]
-        transform = transform_frombuffer_withgrid(minx, maxy, buffer, gsd)
+            sample_bounds = rasterio.coords.BoundingBox(*sample_bounds.bounds)
+        aligned_bounds = pad_grid_align_bounds(sample_bounds, gsd, math.ceil(buffer / gsd))
+        transform = rasterio.transform.from_origin(aligned_bounds.left, aligned_bounds.top, gsd, gsd)
```

Comment:
Since we only used *aligned_bounds* one time as arguments into *rasterio.transform.from_origin()*, we can probably just get away with tuple notation here:
```
transform = rasterio.transform.from_origin(aligned_bounds[0], aligned_bounds[3], gsd, gsd)
```

Reply:
After thinking about these two lines some more, I think this is probably the easier-to-read construction:
```
left, _, _, top = pad_grid_align_bounds(sample_bounds, gsd, math.ceil(buffer / gsd))
transform = rasterio.transform.from_origin(left, top, gsd, gsd)
```